lib/discourse_post_event/event_validator.rb

```diff
@@ -93,7 +93,11 @@ def can_invite_groups?(event)
       return true unless event[:"allowed-groups"]
 
       event[:"allowed-groups"].split(',').each do |group_name|
-        group = Group.find_by(name: group_name)
+        group = begin
+          Group.lookup_group(group_name.to_sym)
+        rescue ArgumentError
+          nil
```

Comment:
this argument error is odd... how can that happen?

Reply:
If there's no automatic or stored group matching the argument (the name) it'll be throw [here](https://github.com/discourse/discourse/blob/main/app/models/group.rb#L616). Rather than throw an error here we want `nil` because then the subsequent conditional will fail returning the right error.